File: agent/src/main/java/io/github/vicitori/threading/highlighter/agent/common/Markers.java
package io.github.vicitori.threading.highlighter.agent.common;

import java.util.List;

/**
 * Registry of all known IntelliJ threading assertion markers.
 */
public final class Markers {

    public static final MarkerInfo SLOW_OPERATION = new MarkerInfo(
            "com.intellij.util.SlowOperations",
            "assertSlowOperationsAreAllowed",
            "Slow Operation",
            "Slow operations are allowed here. This indicates code that may perform I/O or heavy computation."
    );

    public static final MarkerInfo NON_EDT = new MarkerInfo(
            "com.intellij.openapi.application.impl.ApplicationImpl",
            "assertIsNonDispatchThread",
            "Non-EDT Thread",
            "This code must NOT run on the EDT. Background/pooled thread required."
    );

    public static final MarkerInfo EDT = new MarkerInfo(
            "com.intellij.openapi.application.impl.ApplicationImpl",
            "assertIsDispatchThread",
            "EDT Thread",
            "This code must run on the EDT (Event Dispatch Thread). UI operations are allowed."
    );

    private static final List<MarkerInfo> ALL = List.of(SLOW_OPERATION, NON_EDT, EDT);

    public static List<MarkerInfo> getAll() {
        return ALL;
    }

    private Markers() {
    }
}

File: common/src/main/kotlin/io/github/vicitori/threading/highlighter/common/marker/Markers.kt
package io.github.vicitori.threading.highlighter.common.marker

object Markers {
    @JvmField
    val SLOW_OPERATION = MarkerInfo(
        classFqn = "com.intellij.util.SlowOperations",
        methodName = "assertSlowOperationsAreAllowed",
        displayName = "Slow Operation",
        description = "Slow operations are allowed here. This indicates code that may perform I/O or heavy computation."
    )

    @JvmField
    val NON_EDT = MarkerInfo(
        classFqn = "com.intellij.openapi.application.impl.ApplicationImpl",
        methodName = "assertIsNonDispatchThread",
        displayName = "Non-EDT Thread",
        description = "This code must NOT run on the EDT. Background/pooled thread required."
    )

    @JvmField
    val EDT = MarkerInfo(
        classFqn = "com.intellij.openapi.application.impl.ApplicationImpl",
        methodName = "assertIsDispatchThread",
        displayName = "EDT Thread",
        description = "This code must run on the EDT (Event Dispatch Thread). UI operations are allowed."
    )

    @JvmStatic
    fun getAll(): List<MarkerInfo> = listOf(SLOW_OPERATION, NON_EDT, EDT)
}
